Source/Core/Cartesian3.js

```diff
@@ -682,6 +682,24 @@ define([
         return result;
     };
 
+    /**
+     * Projects vector a onto vector b
+     * @param {Cartesian3} a The vector that needs projecting
+     * @param {Cartesian3} b The vector to project onto
+     * @param {Cartesian3} result The result cartesian
+     * @returns {Cartesian3} The modified result parameter
+     */
+    Cartesian3.vectorProjection = function(a, b, result) {
```

Comment:
Can we rename this to `projectVector` or `projectVectors`?

In general, try to use verbs for function names since they are actions.